pkg/common/cns-lib/volume/manager.go

```diff
@@ -476,21 +481,25 @@ func (m *defaultManager) DetachVolume(ctx context.Context, vm *cnsvsphere.Virtua
 		}
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
-			// Volume is already attached to VM
-			diskUUID, err := IsDiskAttached(ctx, vm, volumeID)
-			if err != nil {
-				log.Errorf("DetachVolume: CNS Detach has failed with fault: %q. Unable to check if volume: %q is already detached from vm: %+v",
-					spew.Sdump(volumeOperationRes.Fault), volumeID, vm)
-				return err
-			} else if diskUUID == "" {
-				log.Infof("DetachVolume: volumeID: %q not found on vm: %+v. Assuming volume is already detached",
-					volumeID, vm)
-				return nil
-			} else {
-				msg := fmt.Sprintf("failed to detach cns volume:%q from node vm: %+v. fault: %q, opId: %q", volumeID, vm, spew.Sdump(volumeOperationRes.Fault), taskInfo.ActivationId)
-				log.Error(msg)
-				return errors.New(msg)
+			_, isNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.NotFound)
+			if isNotFoundFault {
+				// check if volume is already detached from the VM
+				diskUUID, err := IsDiskAttached(ctx, vm, volumeID)
+				if err != nil {
+					log.Errorf("DetachVolume: CNS Detach has failed with fault: %q. Unable to check if volume: %q is already detached from vm: %+v",
+						spew.Sdump(volumeOperationRes.Fault), volumeID, vm)
+					return err
+				}
+				if diskUUID != "" {
```

Comment:
In what case will you get `IsNotFound` and `diskUUID != nil`?

Reply:
When FCD is de-registered as CNS Volume, we may get Not Found but FCD could be attached to the Node VM.

Reply:
Can you add that as a comment here?